src/item/Item.php

```diff
@@ -338,6 +357,15 @@ protected function deserializeCompoundTag(CompoundTag $tag) : void{
 		}
 
 		$this->keepOnDeath = $tag->getByte(self::TAG_KEEP_ON_DEATH, 0) !== 0;
+
+		$this->containedItems = [];
+		$containedItems = $tag->getListTag(Container::TAG_ITEMS);
+		if($containedItems !== null && $containedItems->getTagType() === NBT::TAG_Compound){
+			/** @var CompoundTag $itemNBT */
+			foreach($containedItems as $itemNBT){
+				$this->containedItems[] = Item::nbtDeserialize($itemNBT);
+			}
+		}
```

Comment:
I'm not a big fan of this, as its a unique code from container. Wouldnt be possible to move into a child class or somewhere ?

Reply:
Moved the duplicate code to a new `ContainerHelper` class